core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1054,15 +1053,17 @@ class DAGScheduler(
         return
     }
 
+    // Mark the stage as submitted and send an event on the bus. If there are pending tasks, submit
+    // them, otherwise immediately mark the stage as finished, and submit any child stages.
+    stage.latestInfo.submissionTime = Some(clock.getTimeMillis())
```

Comment:
Actually I made a mental note to double-check this later, and neither option will really work because of this code in `JobProgressListener`:

```
          if (stageInfo.submissionTime.isEmpty) {
            // if this stage is pending, it won't complete, so mark it as "skipped":
            skippedStages += stageInfo
```

So the change in semantics I'm introducing is actually wrong, and I'll have to avoid it.